File: trunk/PolarMath/src/org/macondo/polar/evaluation/CV.java
package org.macondo.polar.evaluation;

import org.macondo.polar.data.Training;

/**
 * <p></p>
 *
 * Date: 19.05.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class CV implements Evaluation<Integer> {
    private static final Evaluation<Integer> avg = new Average();
    private static final Evaluation<Integer> sdnn = new SDNN();

    public Integer evaluate(Training training) {
        return 100 * training.evaluate(sdnn) / training.evaluate(avg);
    }
}

File: trunk/MetroApp/HrmMath/Evaluation/HRV/CV.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using HrmMath.Data;
using HrmMath.Evaluation.Statistics;

namespace HrmMath.Evaluation.HRV
{
    public class CV : IEvaluation<double>
    {
        public double Evaluate(SessionData training)
        {
            var sdnn = training.Evaluate(new SDNN());
            var average = training.Evaluate(new Average());
            return sdnn / (double)average * 100;
        }
    }
}
